pipeline/extract.py

```diff
@@ -162,22 +162,13 @@ def parse_pdf(court_case: dict):
     last_page = reader.pages[-1].extract_text()
 
     try:
-        judge = re.search(r"(?<=Before :\n)([A-Z a-z]*)", first_page)
-        if not judge:
-            judge = re.search(r"(?<=Before  : \n \n)([A-Z a-z]*)", first_page)
-        if not judge:
-            judge = re.search(r"(?<=Before : \n \n)([A-Z a-z]*)", first_page)
-        if not judge:
-            judge = re.search(r"(?<=Before:\n)([A-Z a-z]*)", first_page)
-        if not judge:
-            judge = re.search(r"(?<=BEFORE:\n)([A-Z a-z]*)", first_page)
-        if not judge:
-            judge = re.search(r"(THE HONOURABLE [A-Z a-z]*)", first_page)
-        court_case["judge_name"] = judge.group(1).strip()
+        judge = re.search(
+            r"(?:before {0,}: {0,}\n{0,1} {0,1}\n{0,}|the honourable )([a-z .]*)", first_page.lower())
+        court_case["judge_name"] = judge.group(1).strip().replace('.', '')
```

Comment:
I like how you compressed everything into three lines

Reply:
😅